jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -611,6 +628,52 @@ static String getDefaultBaseImage(ProjectProperties projectProperties)
     throw new IncompatibleBaseImageJavaVersionException(11, javaVersion);
   }
 
+  /**
+   * Parses the list of platforms to a set of {@link Platform}.
+   *
+   * @param rawConfiguration raw configuration data
+   * @return the set of parsed platforms
+   * @throws InvalidPlatformConfigurationException if there exists a platformConfiguration in the
+   *     specified platforms list that is missing required fields or has invalid values
+   */
+  @VisibleForTesting
+  static Set<Platform> getPlatformsSet(RawConfiguration rawConfiguration)
+      throws InvalidPlatformConfigurationException {
+    Set<Platform> platforms = new LinkedHashSet<>();
+    for (PlatformConfiguration platformConfiguration : rawConfiguration.getPlatforms()) {
+
+      String platformToString =
+          "architecture="
+              + platformConfiguration.getArchitectureName().orElse("<missing>")
+              + ", os="
+              + platformConfiguration.getOsName().orElse("<missing>");
+
+      if (!platformConfiguration.getArchitectureName().isPresent()) {
+        throw new InvalidPlatformConfigurationException(
+            "platform configuration is missing an architecture value",
+            platformToString,
+            new IllegalArgumentException());
```

Comment:
Is this correct way of throwing a `InvalidPlatformConfigurationException` when I didnt catch any exceptions?

Reply:
We can remove `Throwable cause` from the contructor, as we don't get any exception from another party.
```java
  InvalidPlatformConfigurationException(String message, String platform, Throwable cause) {
```
And for consistency, let's rename the exception class to `InvalidPlatformException`. And when you rename it, you need to update all the Javadocs too.

And because now we are ensure that `getArchitectureName()` and `getOsName()` are always present, there is no need to catch `NoSuchElementException` that will never be thrown. Remove the `try-catch` block below.

Reply:
I ended up changing the constructor